tests/test_filtered_inbox.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) 2022 Genome Research Ltd.
+#
+# Author: Adam Blanchet <ab59@sanger.ac.uk>
+#
+# This file is part of npg_langqc.
+#
+# npg_langqc is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+from typing import List, Tuple
+
+from fastapi.testclient import TestClient
+from ml_warehouse.schema import PacBioRunWellMetrics
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from lang_qc.db.qc_schema import (
+    QcState,
+    QcStateDict,
+    SubProduct,
+    ProductLayout,
+    SeqProduct,
+)
+from lang_qc.endpoints.inbox import (
+    pack_wells_and_states,
+    extract_well_label_and_run_name_from_state,
+    get_well_metrics_from_qc_states,
+)
+from lang_qc.models.inbox_models import FilteredInboxResults
+
+from tests.fixtures.inbox_data import (
+    filtered_inbox_data,
+    inbox_data,
+    wells_and_states,
+)
+
+
+def test_incorrect_filter(test_client: TestClient):
+    """Expect a 422 response when an incorrect filter is given."""
+
+    response = test_client.get("/pacbio/wells?qc_status=thisdoesntexist")
```

Comment:
What happens if there's no qc_status set?

Reply:
Then it should be tested down in `test_inbox_filter`. A casual change of the default behaviour has large ramifications for the client.